src/item.cpp

```diff
@@ -1619,6 +1619,60 @@ bool _stacks_components( item const &lhs, item const &rhs, bool check_components
            ( lhs.get_uncraft_components() == rhs.get_uncraft_components() );
 }
 
+bool _stacks_custom_item_prefix( item const &lhs, item const &rhs )
+{
+    std::string prefix1;
+    for( const flag_id &f : lhs.get_flags() ) {
+        if( !f->item_prefix().empty() ) {
+            prefix1 += f->item_prefix();
+        }
+    }
```

Comment:
Translation can be slow, so you can probably make `item_prefix`/`item_suffix` return `translation` and test emptiness without translating here, and compare the set of flag ids instead, by populating the set like this:

```diff
-    std::string prefix1;
-    for( const flag_id &f : lhs.get_flags() ) {
-        if( !f->item_prefix().empty() ) {
-            prefix1 += f->item_prefix();
-        }
-    }
+    std::set<flag_id> prefix1;
+    for( const flag_id &f : lhs.get_flags() ) {
+        if( !f->item_prefix().empty() ) {
+            prefix1.insert( f );
+        }
+    }
```